src/test/app/XChain_test.cpp

```diff
@@ -0,0 +1,5418 @@
+            while (!claims.empty())
+            {
+                STXChainAttestationBatch attn_batch{
+                    stBridge, claims.begin(), claims.begin() + cnt};
+                auto batch = attn_batch.getJson(JsonOptions::none);
+                auto signer = bridge->signers[signer_idx].account;
+                auto attn_json = xchain_add_attestation_batch(signer, batch);
+                // std::cout << to_string(attn_json) << '\n';
+                env.tx(attn_json);
+                spendFee(signer);
+                claims.erase(claims.begin(), claims.begin() + cnt);
+            }
+        }
+
+        void
+        sendCreateAttestations(
+            size_t signer_idx,
+            BridgeID bridge,
+            CreateClaimVec& claims)
+        {
+            size_t cnt = (claims.size() > 8) ? 8 : claims.size();
+            STXChainBridge const stBridge(bridge->jvb);
+            AttestationBatch::AttestationClaim* nullc = nullptr;
+            while (!claims.empty())
+            {
+                STXChainAttestationBatch attn_batch{
+                    stBridge,
+                    nullc,
+                    nullc,
+                    claims.begin(),
+                    claims.begin() + cnt};
+                auto batch = attn_batch.getJson(JsonOptions::none);
+                auto signer = bridge->signers[signer_idx].account;
+                auto attn_json = xchain_add_attestation_batch(signer, batch);
+                // std::cout << to_string(attn_json) << '\n';
+                env.tx(attn_json);
+                spendFee(signer);
+                claims.erase(claims.begin(), claims.begin() + cnt);
+            }
+        }
+
+        void
+        sendAttestations()
+        {
+            for (size_t i = 0; i < signers_attns.size(); ++i)
+            {
+                for (auto& [bridge, claims] : signers_attns[i])
+                {
+                    sendAttestations(i, bridge, claims.xfer_claims);
+                    sendCreateAttestations(i, bridge, claims.create_claims);
+                }
+            }
+        }
+
+        void
+        init(jtx::Account const& acct)
+        {
+            accounts[acct].init(env, acct);
+        }
+
+        void
+        reinit_accounts()
+        {
+            for (auto& a : accounts)
+                init(a.first);
+        }
+
+        void
+        receive(jtx::Account const& acct, STAmount amt, size_t divisor = 1)
+        {
+            if (amt.issue() != xrpIssue())
+                return;
+            auto it = accounts.find(acct);
+            if (it == accounts.end())
+            {
+                accounts[acct].init(env, acct);
+                it = accounts.find(acct);
+            }
+            it->second.expectedDiff +=
+                (divisor == 1
+                     ? amt
+                     : divide(
+                           amt, STAmount(amt.issue(), divisor), amt.issue()));
```

Comment:
This code is not compiling for me with clang.  I get the following error:
```
/Users/scott/Development/rippled/src/test/app/XChain_test.cpp:5533:33: error: call to constructor of 'ripple::STAmount' is ambiguous
                           amt, STAmount(amt.issue(), divisor), amt.issue()));
                                ^        ~~~~~~~~~~~~~~~~~~~~
/Users/scott/Development/rippled/src/ripple/protocol/STAmount.h:130:5: note: candidate constructor
    STAmount(
    ^
/Users/scott/Development/rippled/src/ripple/protocol/STAmount.h:137:5: note: candidate constructor
    STAmount(
    ^
/Users/scott/Development/rippled/src/ripple/protocol/STAmount.h:143:5: note: candidate constructor
    STAmount(Issue const& issue, std::int64_t mantissa, int exponent = 0);
    ^
/Users/scott/Development/rippled/src/ripple/protocol/STAmount.h:145:5: note: candidate constructor
    STAmount(Issue const& issue, int mantissa, int exponent = 0);
    ^
1 error generated.
```
The problem is that the `divisor` parameter (`size_t`) does not exactly match any of the specific `STAmount` second argument types: `std::uint64_t`, `std::uint32_t`, std::int64_t`, or `int`.  I worked around the problem by changing `divisor`'s type to `std::uint64_t`.  You may want to choose a different value.

Reply:
Thanks you Scott, will fix. 
BTW I updated the `xEnv` class to `XEnv` as you suggested, and split it into a `SEnv` parent class without any prefunded account or definitions, and a `XEnv` class with those that saves much repetition in the tests.

The idea is that people could reuse the `SEnv` class elsewhere if they find it useful. Besides the helper functions, I myself like the fact that calls can be chained, so that you can create an environment, use it, and destroy it within one statement.